packages/cli/src/utils/build.js

```diff
@@ -71,171 +64,398 @@ const requiredFiles = async ({ cwd, entryPoints }) => {
     platform: 'node',
     metafile: true,
     logLevel: 'warning',
-    external: ['../test/userapp'],
+    external: [
+      '../test/userapp',
+      'zapier-platform-core/src/http-middlewares/before/sanatize-headers', // appears in zapier-platform-legacy-scripting-runner/index.js
+      './request-worker', // appears in zapier-platform-legacy-scripting-runner/zfactory.js
+      './xhr-sync-worker.js', // appears in jsdom/living/xmlhttprequest.js
+    ],
     format,
-    conditions,
+    // Setting conditions to an empty array to exclude 'module' condition,
+    // which Node.js doesn't use. https://esbuild.github.io/api/#conditions
+    conditions: [],
```

Comment:
This fixes [PDE-6245](https://zapierorg.atlassian.net/browse/PDE-6245). For more context, see this [Slack message](https://zapier.slack.com/archives/C04P5J5R8E4/p1751015786041659?thread_ts=1750384204.158269&cid=C04P5J5R8E4).